apps/taskmasters/flowlord/taskmaster.go

```diff
@@ -316,9 +316,12 @@ func (tm *taskMaster) Process(t *task.Task) error {
 		} else { // send to the retry failed topic if retries > p.Retry
 			meta.Set("retry", "failed")
 			t.Meta = meta.Encode()
-			if tm.failedTopic != "-" {
+			if tm.failedTopic != "-" && tm.failedTopic != "" {
 				tm.producer.Send(tm.failedTopic, t.JSONBytes())
 			}
+		}
+
+		if t.Result == task.ErrResult || t.Result == task.AlertResult {
```

Comment:
We need to make sure that the error results are only added after being retried. 

``` go 
else { 
    if tm.slack != nil {
	  tm.alerts <- *t
	}
}
if t.Result == task.AlertResult && tm.slack != nil {
	tm.alerts <- *t
}
```

Reply:
that makes sense, I've fixed this.